facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java

```diff
@@ -152,21 +152,22 @@ public T call() throws Exception {
 
     /**
      * Call destination (lightblue) using a timeout in dual read/write phases.
-     * Do not use facade timeout during lightblue proxy phase.
+     * Do not use facade timeout during lightblue proxy and kinda proxy phases (when
+     * reading from source is disabled).
      *
      * @param listenableFuture
      * @param methodName method name is used to read method specific timeout
      * configuration
-     * @param shouldSource true if source operation is supposed to be performed
-     * for this call (i.e. this is a dual phase)
      * @param destinationCallTimeout Set future timeout to this amount
      * @return
      * @throws InterruptedException
      * @throws ExecutionException
      * @throws TimeoutException
      */
-    private <T> T getWithTimeout(ListenableFuture<T> listenableFuture, String methodName, FacadeOperation facadeOperation, boolean shouldSource, int destinationCallTimeout) throws InterruptedException, ExecutionException, TimeoutException {
-        if (!shouldSource || timeoutConfiguration.getTimeoutMS(methodName, facadeOperation) <= 0) {
+    private <T> T getWithTimeout(ListenableFuture<T> listenableFuture, String methodName, FacadeOperation facadeOperation, int destinationCallTimeout) throws InterruptedException, ExecutionException, TimeoutException {
+        // not reading from source/legacy means this is either proxy or kinda proxy phase
+        // in that case, ignore timeout settings for Lightblue call
+        if (!shouldSource(FacadeOperation.READ) || timeoutConfiguration.getTimeoutMS(methodName, facadeOperation) <= 0) {
```

Comment:
If facade recognizes it's proxy or kinda proxy phase, it will call ```listenableFuture.get()``` (no timeout). Otherwise (and when there is a timeout configured), it will call ```listenableFuture.get(destinationCallTimeout, TimeUnit.MILLISECONDS)``` (timeout).